fuel-relayer/src/relayer/get_logs.rs

```diff
@@ -0,0 +1,72 @@
+use super::*;
+use futures::TryStreamExt;
+
+#[cfg(test)]
+mod test;
+
+/// Download the logs from the DA layer.
+pub(crate) fn download_logs<P>(
+    eth_sync_gap: &state::EthSyncGap,
+    contracts: Vec<H160>,
+    eth_node: Arc<P>,
+    page_size: u64,
+) -> impl futures::Stream<Item = Result<Vec<Log>, ProviderError>>
+where
+    P: Middleware<Error = ProviderError> + 'static,
+{
+    // Create a stream of paginated logs.
```

Comment:
Might be helpful to have some logging here to track the relayer download progress:

```suggestion
    // Create a stream of paginated logs.
    `debug!("Downloading logs for block range: {}", &eth_sync_gap);`
```